pkg/orderedmap/convert.go

```diff
@@ -74,7 +74,13 @@ func (c Conversion) fromUnorderedMaps(object interface{}) interface{} {
 			typedObj[i] = c.fromUnorderedMaps(item)
 		}
 		return typedObj
-
+	case []map[string]interface{}:
```

Comment:
Having the case `map[interface{}]interface{}:` earlier in this function makes me think that we should cover the  similar case: `[]map[interface{}]interface{}:`. However, I could not construct a TOML map with non 'string' map keys. 

Ultimately, I was left with this question: Why is the input of an array of maps not covered by the `[]interface{}` case?

Reply:
Another way to ask the question:
- "Why does `toml.decode` return different types of slices rather than always returning `[]interface{}`" 

Found others hitting up against the `toml.Unmarshal` inconsistencies: https://github.com/BurntSushi/toml/issues/169